lib/ansible/modules/cloud/amazon/rds_instance_facts.py

```diff
@@ -0,0 +1,164 @@
+    name: new-database
+  register: new_database_facts
+
+# Get all RDS instances
+- rds_instance_facts:
+'''
+
+RETURN = '''
+instances:
+    description: zero or more instances that match the (optional) parameters
+    type: list
+    returned: always
+    sample:
+       "instances": [
+           {
+               "availability_zone": "ap-southeast-2c",
+               "backup_retention": 10,
+               "create_time": "2017-02-20T06:13:05.724000+00:00",
+               "db_engine": "postgres",
+               "endpoint": "helloworld-rds-master.cyaju3p4tyqv.ap-southeast-2.rds.amazonaws.com",
+               "id": "helloworld-rds-master",
+               "instance_type": "db.t2.micro",
+               "maintenance_window": "sun:15:40-sun:16:10",
+               "multi_zone": false,
+               "port": 5432,
+               "replication_source": null,
+               "size": 5,
+               "status": "available",
+               "username": "hello",
+               "vpc_security_groups": "sg-abcd1234"
+           },
+       ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn, HAS_BOTO3
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+from ansible.module_utils.aws.rds import RDSDBInstance
+
+import traceback
+
+try:
+    import botocore
+except BaseException:
+    pass  # caught by imported HAS_BOTO3
+
+
+def instance_facts(module, conn):
+    instance_name = module.params.get('name')
+    filters = module.params.get('filters')
+
+    params = dict()
+    if instance_name:
+        params['DBInstanceIdentifier'] = instance_name
+    if filters:
+        params['Filters'] = ansible_dict_to_boto3_filter_list(filters)
+
+    marker = ''
+    results = list()
+    while True:
+        try:
+            response = conn.describe_db_instances(Marker=marker, **params)
+            results.extend(response['DBInstances'])
+            marker = response.get('Marker')
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'DBInstanceNotFound':
+                break
+            module.fail_json(msg=str(e), exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+        if not marker:
+            break
+
+    # Get tags for each response
+    for instance in results:
+        instance['tags'] = boto3_tag_list_to_ansible_dict(
+            conn.list_tags_for_resource(
+                ResourceName=instance['DBInstanceArn'])['TagList'])
+
+    module.exit_json(
+        changed=False, db_instances=[
+            RDSDBInstance(instance).data for instance in results])
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(aliases=['instance_name']),
+            filters=dict(type='list', default=[])
```

Comment:
I think this needs to be a dict since it's used as-is in the call to ansible_dict_to_boto3_filter_list().